pkg/model/metric.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2019 Huawei Technologies Co., Ltd. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+This module implements the common data structure.
+*/
+
+package model
+
+type MetricSpec struct {
```

Comment:
Do you need to add `*BaseModel` in MetricSpec?

Reply:
I do not think so. Will re-check once we do the end to end integration and test (from api server till the driver) after the PR's are merged

Reply:
Got it.